dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoRyaInstanceDetailsRepository.java

```diff
@@ -0,0 +1,120 @@
+package mvm.rya.mongodb.instance;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.DB;
+import com.mongodb.DBCollection;
+import com.mongodb.DBCursor;
+import com.mongodb.DBObject;
+import com.mongodb.MongoClient;
+import com.mongodb.WriteResult;
+
+import mvm.rya.api.instance.RyaDetails;
+import mvm.rya.api.instance.RyaDetailsRepository;
+
+/**
+ * An implementation of {@link RyaDetailsRepository} that stores a Rya
+ * instance's {@link RyaDetails} in a Mongo document.
+ */
+@ParametersAreNonnullByDefault
+public class MongoRyaInstanceDetailsRepository implements RyaDetailsRepository {
+    private static final String INSTANCE_DETAILS_COLLECTION_NAME = "instance_details";
+
+    private final DB db;
+    private final String instanceName;
+
+    /**
+     * Constructs an instance of {@link MongoRyaInstanceDetailsRepository}.
+     *
+     * @param client - Connects to the instance of Mongo that hosts the Rya instance. (not null)
+     * @param instanceName - The name of the Rya instance this repository represents. (not null)
+     */
+    public MongoRyaInstanceDetailsRepository(final MongoClient client, final String instanceName) {
+        checkNotNull(client);
+        this.instanceName = requireNonNull( instanceName );
+        db = client.getDB(this.instanceName);
+    }
+
+    @Override
+    public boolean isInitialized() throws RyaDetailsRepositoryException {
+        final DBCollection col = db.getCollection(INSTANCE_DETAILS_COLLECTION_NAME);
```

Comment:
On the Accumulo side I actually fetched the details to see if they were initialized because I was concerned that the table could exist, but have no entry in it.

Have an opinion on that approach?